src/architecture.rs

```diff
@@ -89,4 +89,11 @@ impl Architecture {
             Architecture::Arch128Bit => u128::from_ne_bytes(bytes.try_into().unwrap()) as usize,
         }
     }
+
+    /// Returns the amount of bytes which a pointer takes up on this architecture.
+    /// E.g. 4 for 32bit, 8 for 64bit, etc.
+    #[must_use]
+    pub fn pointer_width_bytes(self) -> usize {
```

Comment:
You can already cast `Architecture` to `usize` with `architecture as usize`. Perhaps what's needed instead is making that availability more clear in documentation?

Reply:
I thought casting an enum was pretty unintuitive, and even if documentation exists but someone stumbles upon it in the wild, they will have no clue what's going on. Especially when that person is new to rust too, and have no idea that it returns the `Arch64Bit = 8` enum value.

I would instead rename `get_pointer_width` to `get_arch`, and keep the function I added (or rename it to `pointer_width`).

Reply:
That's the entire point of the enum being `repr(u8)`, although I guess you'd probably need to read the nomicon to know that.

I'd probably prefer not adding another function, do you think it'd be okay to just add a doc comment to the `Architecture` struct with something along the lines of:
```
This enum is `repr(u8)`, which means that it can be treated exactly the same as a `u8` with the same value as the number of bytes in the architecture.
```